File: backend/Entity/Restaurant.cs
using System.ComponentModel.DataAnnotations.Schema;
using webapi.Base;

namespace backend.Entity
{
    public class Restaurant : BaseCreateDate
    {
        public int Id { get; set; }
        public string? Name { get; set; }
        public double Price { get; set; }
        public string Price_range { get; set; }
        public int? Rating { get; set; }
        public  string? Description { get; set; }
        public string? Address { get; set; }
        public string? Image { get; set; }
        public string? Image_detail { get; set; }
        public int PhoneNumbber { get; set; }
        [ForeignKey(nameof(Location1.ID))]
        public int LocationId { get; set; }
        public Location1? Location { get; set; }
        public string? Links { get; set; }
    }
}

File: frontend/src/app/models/restaurant.ts

export interface Restaurant {
  id?: number;
  name?: string;
  rating?: number | null;
  price_range?: string;
  createDate? : Date;
  description: string;
  image?: string;
  imageDetail?: string;
  address? : string;
  createBy?: string;
  isActive? : boolean;
  intineraries?: any;
  links? :string;
  location1? :any;
  locationId?: number;
  phoneNumber?: number;
  updateBy?: string;
  updateDate?: Date;
}
